dialogue-target/src/main/java/com/palantir/dialogue/ResponseAttachments.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue;
+
+import javax.annotation.Nullable;
+
+public final class ResponseAttachments {
+    private final Attachments attachments = Attachments.create();
+
+    private ResponseAttachments() {}
+
+    public static ResponseAttachments create() {
```

Comment:
This is public because Response implementations are scattered all over.

Reply:
Would be nice if Response was also a concrete type, not sure what we're gaining from it being an interface. If it just took Supplier<InputStream>, code and headers, looks like it should be possible to have a concrete class?

Then this wouldn't have had to be a break I guess. No strong preference though.

Reply:
I think that would result in a subtler break in the case someone wrapped an existing response object, since it would compile but the attachments would be reset.